### PR TITLE
[Docs] Automatic Graphql Docs Generation

### PR DESCRIPTION
**Pull request by dhruv0000**
Signed-off-by: dhruv0000 <patel.4@iitj.ac.in>

**Description**
This PR automatically generates GraphQL docs based on the schema in `internal/graphql/schema/schema.graphql` and creates an custom `.md` page which the Jekyll Site can refer to.

Generate Docs using
```bash
make install
bundle exec rake graphql:compile_docs
```

Watch the generated docs at `extensibility/graphql-reference` endpoint. (temporary)

DONE:
 - Add doc generation capability for:
   - query
   - mutations
   - Object Type
   - Enumeration types
   - Scalar types
   - Input types
   - Abstract types (Commented for future use)
 - Add doc generation capability for:
   - subscriptions
 - Add description for each Object in `schema.graphql`
 - Fix the subsection hierarchy
 - Update the Meshery Server/UI be compatible with the new (minor changes to follow certain GraphQL practices) schema

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by dhruv0000** [comment]
Watch the generated GraphQL docs at https://deploy-preview-2989--meshery-docs.netlify.app/extensibility/graphql-reference

**Comment by leecalcote** [comment]
This is a great looking docs preview.

File: docs/Gemfile.lock [hunk 1]
```diff
@@ -1,6 +1,55 @@
 GEM
   remote: https://rubygems.org/
   specs:
+    actioncable (6.0.3.7)
```
Comment (leecalcote) [inline docs/Gemfile.lock:4]: I hope GitHub Pages supports these. 🤞

**Comment by dhruv0000** [comment]
A review on the GraphQL comments in `internal/graphql/schema/schema.graphql` itself would me much helpful @leecalcote .

**Review by kumarabd — APPROVED** [review]
LGTM

**Comment by kumarabd** [comment]
@dhruv0000 remove `do-not-merge` once ready, we'll merge then

**Comment by dhruv0000** [comment]
Ready for a Merge @abhishekkumkar :+1:

**Review by leecalcote — APPROVED** [review]
Let's see how GH Pages does with this...

Thank you, @dhruv0000!

**Comment by dhruv0000** [comment]
https://docs.meshery.io/extensibility/graphql-reference :rocket:

**Pull request merged**